### PR TITLE
Ruby gem overmind

### PR DESCRIPTION
**Pull request by prog-supdex**
This PR adds the ability to generate gem files that pack the `overmind` library with dependencies `tmux` into a gem file for various architectures of the `Linux` and `macos` platforms by running a rake task `overmind:build:all.`

It also prepares the gem file for `FreeBSD`, but without `tmux` in the box

The rake task will download the overmind and tmux binaries and package them into a platform-specific gem.

To build the gem files, you need to have `Ruby >= 3.0`.

This PR also adds a `NOTICE` file and additional information to the README.md.

### CODE REVIEW

File: packaging/rubygems/lib/overmind/version.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+module Overmind
+  VERSION = "0.1.2"
```
[the remark this answers is not in the document]
Comment (prog-supdex) [reply packaging/rubygems/lib/overmind/version.rb]: done!

File: packaging/rubygems/RELEASING.md
[the remark this answers is not in the document]
Comment (prog-supdex) [reply packaging/rubygems/RELEASING.md:1]: thanks, I added to RELEASING.md

File: packaging/rubygems/Gemfile
[the remark this answers is not in the document]
Comment (prog-supdex) [reply packaging/rubygems/Gemfile:1]: I removed it

File: packaging/rubygems/lib/overmind/cli.rb [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    def run(args = [])
+      os_validate!
+      validate_tmux_present!
+      validate_overmind_present!
+
+      # Ensures arguments are properly quoted if they contain spaces
+      args = args.map { |x| x.include?(" ") ? "'#{x}'" : x }
+
+      # Use prebuild tmux if found
+      path_with_tmux = File.exist?(TMUX_PATH) ? "#{ENV["PATH"]}:#{TMUX_FOLDER_PATH}" : ENV["PATH"]
```
[the remark this answers is not in the document]
Comment (prog-supdex) [reply packaging/rubygems/lib/overmind/cli.rb]: thanks, fixed!

File: NOTICE
[the remark this answers is not in the document]
Comment (prog-supdex) [reply NOTICE:1]: Done!

**Review by Envek — APPROVED** [review]

**Review by DarthSim — APPROVED** [review]

**Pull request merged**